xdsl/transforms/jax_use_donated_arguments.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+
+from xdsl.context import MLContext
+from xdsl.dialects import builtin
+from xdsl.dialects.bufferization import MaterializeInDestination
+from xdsl.dialects.builtin import TensorType
+from xdsl.dialects.func import FuncOp, Return
+from xdsl.ir import Operation, SSAValue
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.utils.exceptions import VerifyException
+
+
+@dataclass
+class SubstituteDonatedTensors(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: Return, rewriter: PatternRewriter, /):
+        func_op = op.parent_op()
+        if func_op is None or type(func_op) is not FuncOp:
```

Comment:
```suggestion
        assert isinstance(func_op, FuncOp)
```
The input can be assumed to be verified. There's no way to handle a VerifyException here anyway, so we might as well assert.